apps/zk-explorer/components/SidebarFilters/SidebarFilters.tsx

```diff
@@ -0,0 +1,206 @@
+import { FC, useEffect, useState } from 'react';
+import { Typography, CheckBox, Chip } from '@webb-tools/webb-ui-components';
+import { PropsOf } from '@webb-tools/webb-ui-components/types';
+import { cloneDeep } from 'lodash';
+import {
+  FilterCategory,
+  FilterCategoryItem,
+  FilterConstraints,
+  FilterOptionItem,
+} from './types';
+
+export type SidebarFiltersProps = PropsOf<'div'> & {
+  onConstraintsChange: (constraints: FilterConstraints) => void;
+};
+
+export const SidebarFilters: FC<SidebarFiltersProps> = ({
+  onConstraintsChange,
+}) => {
+  const [constraints, setConstraints] = useState<FilterConstraints>({
```

Comment:
I'm not sure but this can change to 4 useState hooks to handle each field separately. If you think keeping this is a better option, you can explain it here

Reply:
I was thinking of having them be a single object, since they're always handled together throughout the app; so having 4 would introduce additional complexity, props, and code. What do you think?

Reply:
You can handle the state of these fields with `useSearchParams` of Next.js. Check my previous comment [here](https://github.com/webb-tools/webb-dapp/pull/1902#discussion_r1436773680)